application-licensing-licensor/application-licensing-licensor-api/src/test/java/com/xwiki/licensing/internal/DefaultLicensingConfigurationTest.java

```diff
@@ -20,100 +20,99 @@
 
-import com.xwiki.licensing.LicensingConfiguration;
-
-public class DefaultLicensingConfigurationTest
+@ComponentTest
+class DefaultLicensingConfigurationTest
 {
-    @Rule
-    public MockitoComponentMockingRule<LicensingConfiguration> mocker =
-        new MockitoComponentMockingRule<>(DefaultLicensingConfiguration.class);
+    @InjectMockComponents
+    private DefaultLicensingConfiguration licensingConfiguration;
 
+    @MockComponent
+    @Named("LicensedExtensionAutomaticUpgrades")
     private ConfigurationSource autoUpgradesConfig;
 
-    private ConfigurationSource configurationSource;
-
+    @MockComponent
     private Environment environment;
 
-    @Before
-    public void configure() throws Exception
-    {
-        this.autoUpgradesConfig =
-            this.mocker.getInstance(ConfigurationSource.class, "LicensedExtensionAutomaticUpgrades");
-
-        DefaultParameterizedType configurationSourceProviderType =
-            new DefaultParameterizedType(null, Provider.class, ConfigurationSource.class);
-        Provider<ConfigurationSource> configurationSourceProvider =
-            this.mocker.registerMockComponent(configurationSourceProviderType);
+    @MockComponent
+    private Provider<ConfigurationSource> configurationSourceProvider;
 
-        configurationSource = mock(ConfigurationSource.class);
-        when(configurationSourceProvider.get()).thenReturn(configurationSource);
+    @Mock
+    private ConfigurationSource configurationSource;
 
-        environment = this.mocker.getInstance(Environment.class);
+    @BeforeEach
+    void configure()
+    {
+        when(this.configurationSourceProvider.get()).thenReturn(this.configurationSource);
     }
 
     @Test
-    public void getBlocklist() throws Exception
+    void getAllowList() throws Exception
     {
         // Since getProperty method returns a list of objects, we check also that the conversion to string is done
         // correctly.
-        List<Object> blocklist = Arrays.asList(1, 2, null);
+        List<Object> allowlist = Arrays.asList(1, 2, null);
 
-        when(this.autoUpgradesConfig.getProperty("blocklist")).thenReturn(blocklist);
+        when(this.autoUpgradesConfig.getProperty("allowlist")).thenReturn(allowlist);
 
-        assertEquals(Arrays.asList("1", "2", null), mocker.getComponentUnderTest().getAutoUpgradeBlocklist());
+        assertEquals(Arrays.asList("1", "2", null), this.licensingConfiguration.getAutoUpgradeAllowList());
     }
 
-    @Test(expected = RuntimeException.class)
-    public void getBlocklistWithException() throws Exception
+    @Test
+    void getAllowListWithException() throws Exception
     {
-        when(this.autoUpgradesConfig.getProperty("blocklist")).thenReturn("not a list");
-        mocker.getComponentUnderTest().getAutoUpgradeBlocklist();
+        try {
+            when(this.autoUpgradesConfig.getProperty("allowlist")).thenReturn("not a list");
+            this.licensingConfiguration.getAutoUpgradeAllowList();
```

Comment:
You're missing a ``fail()`` call after this line, to fail the test is no exception is thrown.